plugin/kubernetes/handler.go

```diff
@@ -53,6 +57,48 @@ func (k Kubernetes) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.M
 			break
 		}
 		fallthrough
+	case dns.TypeAXFR:
+		records, err = plugin.AXFR(&k, zone, state, plugin.Options{})
+		if k.Fallthrough {
+			fakew := new(fakewriter)
+			fakew.RemoteIP = w.RemoteAddr()
+
+			_, err := plugin.NextOrFailure(k.Name(), k.Next, ctx, fakew, r)
+			if err != nil {
+				break
+			}
+
+			fakeMsg := fakew.ReadMsg()
+			// Strip out SOA records
+			records = append(records, fakeMsg.Answer[1:len(fakeMsg.Answer)-1]...)
```

Comment:
I don't understand what's happening here... Why the `if k.Fallthrough`? and why the `fakewriter`

Maybe it neater rename plugin.Xfr to plugin.All() or have a private function that just grabs all records.

Reply:
If fall through is set, then the k8s plugin does not know all of the records that are in the zone. Some other later plugin may be handling some records. So, we push a request down the plugin chain to gather up all those records that may be managed by someone else.

Reply:
Yeah, what John said.
Our use case is to have host records at `cluster.local`, but then k8s services at `xxx.namespace.svc.cluster.local`. So we use kubernetes and then a fallthrough to file based for host entries. 
By using the `fakewriter` we can let the next plugin write out results locally and then inject the data into the payload we send back to the client.

Reply:
hmm, I don't like the complexity this introduces. I thought it would be enough to have it just return the records it has in the cluster - not grabbing things from all over the place?

Reply:
Well, if we are doing a zone transfer shouldn't we include all the records in that zone, regardless of which plugin handles them? From the point of view of an external server, it doesn't care how we construct the records internally, it just needs to know that it has everything for the requested zone.

I don't think this is really any more complex than fall through in general - and allows help debugging fall through results.

Reply:
No, we should do this Fallthrough crazyness. In fact it is already leading to weird to debug errors.

Reply:
What errors? We should go ahead without it for this PR and then add another PR to do it, and discuss it there. To me, it seems we are responsible for making sure that the AXFR contains all the records we might return for the zone.